warehouse/query-core/src/main/java/datawave/query/jexl/visitors/UnmarkedBoundedRangeDetectionVisitor.java

```diff
@@ -22,26 +34,30 @@ public static boolean findUnmarkedBoundedRanges(JexlNode script) {
     }
 
     @Override
-    public Object visit(ASTReference node, Object data) {
-        // determine if we have a marked range that is not actually a range
-        if (QueryPropertyMarker.findInstance(node).isType(BOUNDED_RANGE)) {
-            if (null != data && !JexlASTHelper.findRange().isRange(node)) {
+    public Object visit(ASTAndNode node, Object data) {
+        if (data == null) {
+            return null;
+        }
+
+        // check for a bounded marker where the source node is not a range
+        Instance instance = QueryPropertyMarker.findInstance(node);
+        if (instance.isType(BOUNDED_RANGE)) {
+            LiteralRange<?> range = JexlASTHelper.findRange().getRange(node);
+            if (range == null) {
                 AtomicBoolean hasBounded = (AtomicBoolean) data;
                 hasBounded.set(true);
             }
-
             return false;
         }
-        // determine if we have a range that is not marked
-        else if (JexlASTHelper.findRange().notDelayed().notMarked().isRange(node)) {
-            if (null != data) {
-                AtomicBoolean hasBounded = (AtomicBoolean) data;
-                hasBounded.set(true);
-            }
 
+        // check for a range that is not marked
+        LiteralRange<?> range = JexlASTHelper.findRange().notDelayed().notMarked().getRange(node);
+        if (range != null && range.isBounded()) {
+            AtomicBoolean hasBounded = (AtomicBoolean) data;
```

Comment:
if its not marked is there any concern about the data type cast?

Reply:
No. But now I'm leaning towards a larger refactor for clarity.